app/src/main/java/emu/skyline/preference/FilePreference.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+package emu.skyline.preference
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.util.AttributeSet
+import androidx.preference.Preference
+import androidx.preference.R
+
+class FilePreference @JvmOverloads constructor(context : Context?, attrs : AttributeSet? = null, defStyleAttr : Int = R.attr.preferenceStyle) : Preference(context, attrs, defStyleAttr), ActivityResultDelegate {
+
+    private var requestCode = 0
+
+    override fun onClick() = (context as Activity).startActivityForResult(Intent(context, FileActivity::class.java).apply { putExtra(DocumentActivity.KEY_NAME, key) }, requestCode)
+
+    override fun assignActivityRequestCode(requestCode : Int) {
+        this.requestCode = requestCode
+    }
+
+    override fun onActivityResult(requestCode : Int, resultCode : Int, data : Intent?) {
+    }

```

Comment:
You should have the default implementation in the base class and just not at all override it here, once you move `requestCode` into the base class this class will be nearly empty